packages/hardhat-chai-matchers/test/bigNumber.ts

```diff
@@ -864,6 +864,64 @@ describe("BigNumber matchers", function () {
         });
       });
     });
+
+    describe("deep equal", async function () {
+      checkAll(1, 1, (a, b) => {
+        it(`should work with ${typestr(a)} and ${typestr(b)}`, function () {
+          // successful assertions
+          expect([a]).to.deep.equal([b]);
+          expect([[a], [a]]).to.deep.equal([[b], [b]]);
+          expect({ x: a }).to.deep.equal({ x: b });
+          expect({ x: { y: a } }).to.deep.equal({ x: { y: b } });
+          expect({ x: [a] }).to.deep.equal({ x: [b] });
+
+          // failed assertions
+          expect(() => expect([a]).to.not.deep.equal([b])).to.throw(
+            AssertionError
```

Comment:
Agree that we should have some error message checking here. I'm not sure what's the best way to do it because the way the different numbers are serialized might differ, but I'll give it a try.